runtime/src/governance/election.rs

```diff
@@ -651,11 +646,9 @@ impl<T: Trait> Module<T> {
 
         let vote_stake = Self::new_stake_reusing_transferable(&mut transferable_stake.backing, stake);
 
-        let balance = <balances::Module<T>>::free_balance(&voter);
-
-        ensure!(balance >= vote_stake.new, "not enough balance to cover voting stake");
+        ensure!(<balances::Module<T>>::can_reserve(&voter, vote_stake.new), "not enough free balance to reserve");
 
-        ensure!(<balances::Module<T>>::decrease_free_balance(&voter, vote_stake.new).is_ok(), "failed to update balance");
+        ensure!(<balances::Module<T>>::reserve(&voter, vote_stake.new).is_ok(), "failed to reserve voting stake!");
```

Comment:
The same here:
`<balances::Module<T>>::reserve(&voter, vote_stake.new)?;`

Reply:
You can even transform a returned error in this way:
`<balances::Module<T>>::reserve(&voter, vote_stake.new).map_err(|_| "failed to reserve voting stake!")?;`